src/SDK/Resource/Detectors/Host.php

```diff
@@ -15,13 +15,128 @@
  */
 final class Host implements ResourceDetectorInterface
 {
+    private const PATH_ETC_MACHINEID = 'etc/machine-id';
+    private const PATH_VAR_LIB_DBUS_MACHINEID = 'var/lib/dbus/machine-id';
+    private const PATH_ETC_HOSTID = 'etc/hostid';
+    private readonly string $dir;
+    private readonly string $os;
+
+    public function __construct(string $dir = '/', string $os = PHP_OS_FAMILY)
+    {
+        $this->dir = $dir;
+        $this->os = $os;
+    }
+
     public function getResource(): ResourceInfo
     {
         $attributes = [
             ResourceAttributes::HOST_NAME => php_uname('n'),
             ResourceAttributes::HOST_ARCH => php_uname('m'),
+            ResourceAttributes::HOST_ID => $this->getMachineId(),
         ];
 
         return ResourceInfo::create(Attributes::create($attributes), ResourceAttributes::SCHEMA_URL);
     }
+
+    private function getMachineId()
+    {
+        switch (strtolower($this->os)) {
+            case 'linux':
+                {
+                    return $this->getLinuxId();
+                }
+            case 'freebsd':
+            case 'netbsd':
+            case 'openbsd':
+                {
+                    return $this->getBsdId();
+                }
+            case 'darwin':
+                {
+                    $out = $this->getMacOsId();
+
+                    return self::parseMacOsId($out);
+                }
+            case 'windows':
+                {
+                    $out = $this->getWindowsId();
+
+                    return self::parseWindowsId($out);
+                }
+        }
+
+        return '';
+    }
+
+    private function getLinuxId(): string
+    {
+        $paths = [self::PATH_ETC_MACHINEID, self::PATH_VAR_LIB_DBUS_MACHINEID];
+
+        foreach ($paths as $path) {
+            if (file_exists($this->dir . $path)) {
+                return trim(file_get_contents($this->dir . $path));
+            }
+        }
+
+        return '';
+    }
+
+    private function getBsdId(): string
+    {
+        if (file_exists($this->dir . self::PATH_ETC_HOSTID)) {
+            return trim(file_get_contents($this->dir . self::PATH_ETC_HOSTID));
+        }
+
+        $out = exec('kenv -q smbios.system.uuid');
+
+        if ($out != false) {
+            return $out;
+        }
+
+        return '';
+    }
+
+    private function getMacOsId(): string
+    {
+        $out = exec('ioreg -rd1 -c "IOPlatformExpertDevice"');
+
+        if ($out != false) {
+            return $out;
+        }
+
+        return '';
+    }
+
+    private function getWindowsId(): string
+    {
+        $out = exec('%windir%\System32\REG.exe QUERY HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography /v MachineGuid');
```

Comment:
Could use [`Get-ItemPropertyValue`](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-itempropertyvalue?view=powershell-7.4#example-1-get-the-value-of-the-productid-property) to replace `::parseWindowsId()`.
```suggestion
        $out = exec('powershell.exe -Command "Get-ItemPropertyValue -Path HKLM:\SOFTWARE\Microsoft\Cryptography -Name MachineGuid"');
```